include/libopencm3/pac55xx/usart.h

```diff
@@ -0,0 +1,225 @@
+/** LCR:PSEL & LCR:PEN Odd parity */
+#define USART_PSELPEN_ODD       (0b001)
+/** LCR:PSEL & LCR:PEN Even parity */
+#define USART_PSELPEN_EVEN      (0b011)
+/** LCR:PSEL & LCR:PEN Force 1 stick parity */
+#define USART_PSELPEN_FORCE1    (0b101)
+/** LCR:PSEL & LCR:PEN Force 0 stick parity */
+#define USART_PSELPEN_FORCE0    (0b111)
+/** LCR:PSEL & LCR:PEN Disable parity */
+#define USART_PARITY_DISABLE    (0b000)
+/** LCR:PSEL & LCR:PEN Odd parity */
+#define USART_PARITY_ODD        USART_PSELPEN_ODD
+/** LCR:PSEL & LCR:PEN Even parity */
+#define USART_PARITY_EVEN       USART_PSELPEN_EVEN
+/** LCR:PSEL & LCR:PEN Force 1 stick parity */
+#define USART_PARITY_FORCE1     USART_PSELPEN_FORCE1
+/** LCR:PSEL & LCR:PEN Force 0 stick parity */
+#define USART_PARITY_FORCE0     USART_PSELPEN_FORCE0
+/** LCR:SBS Use 1 stop bit */
+#define USART_STOPBITS_1        (0)
+/** LCR:SBS Use 1.5 stop bit when databits is 5 */
+#define USART_STOPBITS_1P5      USART_LCR_SBS
+/** LCR:SBS Use 2 stop bits */
+#define USART_STOPBITS_2        USART_LCR_SBS
+#define USART_LCR_WLS_MASK      (0b11)
+/** Word length select: 5-8 databits */
+#define USART_LCR_WLS(wls)      ((wls) & USART_LCR_WLS_MASK)
+/** Set LCR:SBS for 1.5 or 2 stop bits, Clear for 1 stop bit */
+#define USART_LCR_SBS           BIT2
+/** Enable parity checking */
+#define USART_LCR_PEN           BIT3
+#define USART_LCR_PSELPEN_MASK  (0b0111)
+#define USART_LCR_PSELPEN_SHIFT 3
+/** LCR:PSEL and LCR:PEN control parity */
+#define USART_LCR_PSELPEN(psel) (((psel) & USART_LCR_PSELPEN_MASK) << USART_LCR_PSELPEN_SHIFT)
+/** Break Control: Enabling this bit forces TX to logic 0 */
+#define USART_LCR_BCON          BIT6
+/**@}*/
+
+/** @defgroup usart_lsr_bits Line Status Register bits
+@{*/
+/** Receiver Data Ready */
+#define USART_LSR_RDR           BIT0
+/** Overrun Error */
+#define USART_LSR_OE            BIT1
+/** Parity Error */
+#define USART_LSR_PE            BIT2
+/** Framing Error */
+#define USART_LSR_FE            BIT3
+/** Break Interrupt */
+#define USART_LSR_BI            BIT4
+/** Transmitter Holding Register Empty */
+#define USART_LSR_THRE          BIT5
+/** Transmitter Empty */
+#define USART_LSR_TEMT          BIT6
+/** Error in RX FIFO */
+#define USART_LSR_RXFE          BIT7
+/**@}*/
+
+/** TX FIFO depth */
+#define USART_TX_FIFO_DEPTH     (16)
+/** RX FIFO depth */
+#define USART_RX_FIFO_DETPH     (16)
+
+/** Enable Enhanced Mode to use TX and RX FIFO trigger level interrupts */
+#define USART_EFR_ENMODE        BIT4
+
+/**@}*/
+
+BEGIN_DECLS
+
+uint32_t usart_set_baudrate(uint32_t usart, uint32_t baud);
+void usart_configure_lcr(uint32_t usart, uint8_t data_bits, uint8_t stop_bits,
+				uint8_t parity);
+void usart_break_enable(uint32_t usart);
+void usart_break_disable(uint32_t usart);
+void usart_enhanced_enable(uint32_t usart);
+void usart_enhanced_disable(uint32_t usart);
+bool usart_configure_fcr(uint32_t usart, uint8_t tx_depth, uint8_t rx_depth);
```

Comment:
would usart_set_fifo_depth() not be easier to understand?

Reply:
I think it probably would be easier to understand.  So, we could have usart_set_fifo_depth() first call usart_fifo_enable(), then this wouldn't need a return value.  Is that the cleanest way you all see @karlp @vielster ?

Reply:
That sounds okay to me, I would just suggest it is documented that the call will enable FIFOs if they're not already.